ipa_room_exploration/ros/src/room_exploration_action_server.cpp

```diff
@@ -433,7 +433,8 @@ void RoomExplorationServer::exploreRoom(const ipa_building_msgs::RoomExploration
 			VoronoiMap vm(room_gridmap.data.data(), room_gridmap.info.width, room_gridmap.info.height, grid_spacing_as_int); // a perfect alignment of the paths cannot be assumed here (in contrast to footprint planning) because the well-aligned fov trajectory is mapped to robot locations that may not be on parallel tracks
 			// get the exploration path
 			std::vector<geometry_msgs::Pose2D> fov_path_uncleaned;
-			vm.generatePath(fov_path_uncleaned, cv::Mat());
+			vm.setSingleRoom(true); //to force to consider all rooms
+			vm.generatePath(fov_path_uncleaned, cv::Mat(), room_gridmap.info.width/2,room_gridmap.info.height/2 /*start position in mid*/);
```

Comment:
Is this `room_gridmap.info.width/2` meant to have the robot start in the center of the room? Sometimes this point is not part of the room, so could we just use `starting_position.x` and `starting_position.y` instead which are computed to be as central as possible and accessible in the room?